apps/mark/frontend/src/app_root.tsx

```diff
@@ -619,6 +630,32 @@ export function AppRoot({
     void updateStorage();
   }, [storage]);
 
+  const [scannerReportDataToBePrinted, setScannerReportDataToBePrinted] =
+    useState<ScannerReportData>();
+
+  useQueryChangeListener(
+    authStatusQuery,
+    (
+      newAuthStatus,
+      previousAuthStatus = InsertedSmartCardAuth.DEFAULT_AUTH_STATUS
+    ) => {
+      if (
+        !isPollWorkerAuth(previousAuthStatus) &&
+        isPollWorkerAuth(newAuthStatus)
+      ) {
+        readScannerReportDataFromCardMutation.mutate(undefined, {
+          onSuccess(result) {
+            if (result.isOk() && result.ok() !== undefined) {
+              setScannerReportDataToBePrinted(result.ok());
+            }
+          },
+        });
+      } else if (!isPollWorkerAuth(newAuthStatus)) {
+        setScannerReportDataToBePrinted(undefined);
+      }
+    }
+  );
```

Comment:
Now that I'm seeing this again, I'm wondering why reading the report data is a mutation rather than a query. I should have asked before but it didn't occur to me until now.

Reply:
Mm that is a good catch - a query (with no caching so that we don't have to worry about clearing when a card is removed) probably makes more sense. Let me update

Reply:
Required some thinking / exploration of the react-query API but got things working!